### PR TITLE
BaSP-TG-W08: super-admin

### PR DESCRIPTION
**Pull request by julianpiaggio**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 // use "import" to import libraries
 import express from 'express';
+import {
+  superAdminAlls, getSuperAdminById, superAdminCreate, deleteSuperAdmin, editSuperAdmin,
+  filterAdmin,
```
Comment (kolodani) [inline src/index.js:5]: mas que barbaro barba, aca podrias acomodarlo como en el archivo que lo importas, uno por linea o por renglon

File: src/resources/super-admins.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      res.status(404).json({ error: 'Cannot delete super admin' });
+    } else {
+      res.status(200).json({ data: deleted });
+    }
+  });
+};
+
+const editSuperAdmin = (req, res) => {
+  const element = parseInt(req.params.id, 10);
+  const editSA = superAdmins.find((superAdmin) => superAdmin.id === element);
```
Comment (leimeter-joaquin) [inline src/resources/super-admins.js:44]: Should be named SAToEdit or just superAdmin. Like this it sounds like a function/action.

File: src/resources/super-admins.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if (!deleted) {
+      res.status(404).json({ error: 'Cannot delete super admin' });
+    } else {
+      res.status(200).json({ data: deleted });
+    }
+  });
+};
+
+const editSuperAdmin = (req, res) => {
+  const element = parseInt(req.params.id, 10);
```
Comment (leimeter-joaquin) [inline src/resources/super-admins.js:43]: SuperAdminId seems like a better name for this function.

File: src/resources/super-admins.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const editSA = superAdmins.find((superAdmin) => superAdmin.id === element);
+
+  editSA.id = req.body.id;
+  editSA.name = req.body.name;
+  editSA.lastName = req.body.lastName;
+  editSA.email = req.body.email;
+  editSA.password = req.body.password;
```
Comment (leimeter-joaquin) [inline src/resources/super-admins.js:50]: Do a check for each property. If the property is not sent, we should not change it.

File: src/resources/super-admins.js (hunk 3 above)
Comment (leimeter-joaquin) [reply src/resources/super-admins.js:43]: Dissmiss this comment, I'm wrong

**Review by leimeter-joaquin — APPROVED** [review]

**Review by mara666 — APPROVED** [review]
Great job, congrats!

**Pull request merged**